app/src/main/java/org/simple/clinic/appupdate/criticalupdatedialog/CriticalAppUpdateDialog.kt

```diff
@@ -0,0 +1,163 @@
+package org.simple.clinic.appupdate.criticalupdatedialog
+
+import android.content.ActivityNotFoundException
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.View.GONE
+import android.view.View.VISIBLE
+import android.view.ViewGroup
+import com.jakewharton.rxbinding3.view.clicks
+import com.spotify.mobius.functions.Consumer
+import io.reactivex.Observable
+import io.reactivex.rxkotlin.cast
+import kotlinx.parcelize.Parcelize
+import org.simple.clinic.ContactType
+import org.simple.clinic.R
+import org.simple.clinic.ReportAnalyticsEvents
+import org.simple.clinic.appupdate.AppUpdateNudgePriority
+import org.simple.clinic.databinding.DialogCriticalAppUpdateBinding
+import org.simple.clinic.di.injector
+import org.simple.clinic.navigation.v2.ScreenKey
+import org.simple.clinic.navigation.v2.fragments.BaseDialog
+import org.simple.clinic.platform.crash.CrashReporter
+import org.simple.clinic.util.UserClock
+import org.simple.clinic.widgets.UiEvent
+import java.time.LocalDate
+import javax.inject.Inject
+
+class CriticalAppUpdateDialog : BaseDialog<
+    CriticalAppUpdateDialog.Key,
+    DialogCriticalAppUpdateBinding,
+    CriticalAppUpdateModel,
+    CriticalAppUpdateEvent,
+    CriticalAppUpdateEffect,
+    CriticalAppUpdateViewEffect>(),
+    CriticalAppUpdateUi, UiActions {
+
+  companion object {
+    const val PLAY_STORE_URL_FOR_SIMPLE = "https://play.google.com/store/apps/details?id=org.simple.clinic"
```

Comment:
I think we are using this URL in multiple places. Should we just have a `Global` variable for this? WDYT?

```kt
val simplePlayStoreLink = "https://play.google.com/store/apps/details?id=${BuildConfig.APPLICATION_ID}"
```